Formula/a/apache-arrow.rb

```diff
@@ -55,6 +45,8 @@ def install
     args = %W[
       -DCMAKE_INSTALL_RPATH=#{rpath}
       -DLLVM_ROOT=#{Formula["llvm"].opt_prefix}
+      -DCMAKE_C_COMPILER=#{Formula["llvm"].opt_bin}/clang
+      -DCMAKE_CXX_COMPILER=#{Formula["llvm"].opt_bin}/clang++
```

Comment:
macOS failure is likely due to this as it is incorrectly bypassing brew shim.  
```suggestion
```

Also Linux build likely needs https://github.com/apache/arrow/commit/00245cc802bc3be9a9cd169017f285586483fbb5